docs/blog/posts/2023/sr-linux-kubernetes-anycast-lab.md

```diff
@@ -0,0 +1,955 @@
+
+Wouldn't it be great to combine both worlds?
+
+In this blog post we will explore a lab topology consisting of a Leaf/Spine [SR Linux](https://learn.srlinux.dev/) Fabric connected to a Kubernetes Cluster.
+
+Our k8s Cluster will feature [MetalLB](https://metallb.universe.tf/), which is a load-balancer implementation for bare metal clusters. This will unlock the possibility to have anycast services in our fabric.
+
+<!-- more -->
+
+## Lab summary
+
+| Summary                   |                                                                                                                                                                                                |
+| ------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| **Lab name**              | SR Linux Kubernetes Anycast Lab                                                                                                                                                                        |
+| **Lab components**        | Nokia SR Linux, Kubernetes, MetalLB                                                                                                                                                            |
+| **Resource requirements** | :fontawesome-solid-microchip: 6 vCPU <br/>:fontawesome-solid-memory: 16 GB                                                                                                                      |
+| **Lab**                   | [srl-labs/srl-k8s-anycast-lab][lab]                                                                                                                                                               |
+| **Version information**   | [`containerlab:0.42.0`](https://containerlab.dev/install/), [`srlinux:23.3.3`](https://github.com/nokia/srlinux-container-image),[`minikube v1.30.1`](https://minikube.sigs.k8s.io/docs/start/)|
+| **Authors**               | Míchel Redondo [:material-linkedin:][rd-linkedin]                                                                                                                                              |
+
+At the end of this blog post you can find a quick summary of the steps ([TL;DR](#tldr)).
+
+## Prerequisites
+
+The lab leverages [Containerlab](https://containerlab.dev/)  to spin up a Leaf/Spine Fabric coupled with [Minikube](https://minikube.sigs.k8s.io/) to deploy the Kubernetes cluster.
+
+[Docker engine](https://docs.docker.com/engine/install/) has to be installed on the host system.
+
+## Lab topology
+
+The goal of this lab is to provide users with an environment to test the network integration of a Kubernetes cluster with a Leaf/Spine SR Linux fabric.   
+
+
+<center markdown>![topology](sr-linux-kubernetes-anycast-lab/topology.svg){width=100%}</p>
+</small>Topology</small></center>
```

Comment:
@michelredondo 
the way we deal with drawio graphics is that we source it from the github repo using the JS wrapper to enable live edits to the source file to be rendered in the blog post without rebuilding.

This requires to add the following line to a page - https://github.com/srl-labs/learn-srlinux/blob/66cf0dcc56b02722b91a1659340a8e55cea6ba92/docs/tutorials/programmability/ansible/intent-based-management/config.md#L617 and add a link to the drawing like this https://github.com/srl-labs/learn-srlinux/blob/66cf0dcc56b02722b91a1659340a8e55cea6ba92/docs/tutorials/programmability/ansible/intent-based-management/config.md#L528

Also, to support both day/night themes and enhance reading experience, our drawings have a grey backdrop. I am afraid you'll have to adapt your drawing so that it is visible using the night theme.

Reply:
Also you have three leaves, but only 3 worker nodes. This looks weird.
Maybe use three leaves?

Reply:
The first three clients (connected to leaf,leaf2,leaf3) have directly connected k8s nodes, no ECMP here.
The fourth client (leaf4), does not have a directly connected k8s node. He will see the  three k8s nodes in an ECMP set.

Reply:
ah I see, the topology drawing is misleading then.
It should somehow show that leaves are connected to both clients and k8s nodes.
Currently it shows a single link to a bubble of k8s, which adds confusion